modules/ROOT/pages/mule-error-concept.adoc

```diff
@@ -60,6 +60,36 @@ All errors are either general or `CRITICAL`, the latter being so severe that the
 
 When it comes to connectors, each connector defines its error type hierarchy considering the core runtime hierarchy, though `CONNECTIVITY` and `RETRY_EXHAUSTED` types are always present since they are common to all connectors.
 
+Below is listed a full list of Error Types:
+
+* `ANY`: Error type that matches all error types that may occur in an Flow, that can be handled. This does not include errors that may occur on the source. 
+** `TRANSFORMATION`: indicates an error ocurred while transforming a value.
+** `EXPRESSION`: indicates an error ocurred while evaluating an expression.
+** `VALIDATION`: indicates a validation error occured.
+*** `DUPLICATE_MESSAGE`: indicates a validation error regarding a message being processed twice. For example, using the xref:idempotent-message-validator.adoc[Idempotent message validator].
+** `REDELIVERY_EXHAUSTED`: indicates that max attemps to reprocess a message from a source has been exhausted.
+** `CONNECTIVITY`: indicates that there was a problem establishing a connection. For example, any connector. 
+*** `RETRY_EXHAUSTED`: indicates that retries of a certain execution block has been exhausted. For example, for a given extension, or using xref:until-successful-scope.adoc[Until Successful Scope].
+** `ROUTING`: indicates an error occured while routing a message. For example, using the xref:round-robin.adoc[Round Robin router].
+*** `COMPOSITE_ROUTING`: indicates that one or more errors occured during routing a message. For example, using a xref:scatter-gather-concept.adoc[Scatter Gather Router].
+** `SECURITY`: indicates a security error ocurred, like invalid credentials or expired token. 
+*** `CLIENT_SECURITY`: indicates a security error occured due to an external entity, such as calling an external endpoint. 
```

Comment:
This is more an external entity producing a security error, so an operation is rejected.